### ISSUE TITLE
Cleanup code

### ISSUE THREAD

**Issue by araffin** [DLR-RM/rl-baselines3-zoo#5]
+ use f-strings everywhere
+ type hints

**Comment by salmannotkhan** [comment]
Well I want to help. I don't know ML but I can refactor the code and make it clean.

**Comment by araffin** [comment]
>Well I want to help. I don't know ML but I can refactor the code and make it clean.

That would be great =) Maybe do it part by part (so we can agree on how to do it ;) )
It's also about typing and making it follow pep8. I will setup Github workflow with flake8 check soon (as I did for SB3).

**Comment by salmannotkhan** [comment]
so for now can I start refactoring?

**Comment by araffin** [comment]
>so for now can I start refactoring?

yes you can, but please create a draft PR when you have one file refactored ;)

**Comment by salmannotkhan** [comment]
You're already using f-string in train.py

**Comment by araffin** [comment]
>You're already using f-string in train.py

Everywhere? The main issue currently is that I did half a cleanup. So, some methods are typed, some are using f-strings, some are documented but not everything...

**Comment by salmannotkhan** [comment]
Yeah I found some dirty string inside utils cleaning up now :)

**Comment by salmannotkhan** [comment]
And there is one more thing some strings are double-quoted and some strings are single-quoted, should i fix those too??

**Comment by salmannotkhan** [comment]
`try:
            x, y = ts2xy(load_results(reward_log), 'timesteps')
            skip_eval = len(x) > 0
        except:
            pass`
which type of exception this try block throws?

**Comment by araffin** [comment]
>And there is one more thing some strings are double-quoted and some strings are single-quoted, should i fix those too?

Yes, you can

> which type of exception this try block throws?

Good question. I would say it may fail for several reasons. It tries to load a csv file and may be fail because:
- the file does not exist
- the file is not properly formatted
- the files does contain only headers

unfortunately, I wrote that piece of code [long time ago](https://github.com/araffin/rl-baselines-zoo/commit/af7fc6c65a3478f8ad6a7e911856569985340258) (already two years!) so I don't remember exactly what time of exception was thrown...

**Comment by salmannotkhan** [comment]
> Good question. I would say it may fail for several reasons. It tries to load a csv file and may be fail because:

> the file does not exist
> the file is not properly formatted
> the files does contain only headers
>

I'm asking because it's bare except which is not ideal in python

**Comment by salmannotkhan** [comment]
I created draft PR. Check it out.

**Comment by araffin** [comment]
>I'm asking because it's bare except which is not ideal in python

I know, that's why I created that issue... I would need to make it crash again to give you an answer

**Comment by salmannotkhan** [comment]
> I know, that's why I created that issue... I would need to make it crash again to give you an answer
Yes, Because I don't know how to test this thing.

**Comment by araffin** [comment]
>Yes, Because I don't know how to test this thing.

It's in the readme (once the dependencies are installed):
`python -m utils.benchmark`
but to re-create the errors, you would need to create corrupted files... Don't bother too much about that one.

**Comment by salmannotkhan** [comment]
Okay. Check my PR if you approve I can refactor other files.

**Comment by salmannotkhan** [comment]
why we're importing tkinter in benchmark.py?

**Comment by salmannotkhan** [comment]
Hey, I'm refactoring callbacks.py
Why we're importing pickle?

**Comment by salmannotkhan** [comment]
I created another PR for record_video.py. Please verify it.

**Comment by salmannotkhan** [comment]
I have one question what to do about complexity warnings?

**Comment by araffin** [comment]
>What to do about complexity warnings?

You can ignore them for now

>Why we're importing pickle in callbacks.py?

no good reason anymore

**Comment by salmannotkhan** [comment]
> no good reason anymore

So, can i remove it?

**Comment by araffin** [comment]
yes, please do, I added `# pytype: disable=import-error` to the ones that should stay ;)

**Comment by salmannotkhan** [comment]
OKAY

**Comment by salmannotkhan** [comment]
and what about assigned but never used variables in hyperparams?

**Comment by salmannotkhan** [comment]
![Screenshot_20200519_222920](https://user-images.githubusercontent.com/26996284/82355696-3100fa00-9a20-11ea-9292-bfd0004c7614.png)
can you explain why the assert has these lines?

**Comment by salmannotkhan** [comment]
in train.py instead of writing all login in if statement, we can create a function with that code and then call that function in if statement just to reduce complexity

**Comment by salmannotkhan** [comment]
In `test_hyperparams_opt.py` `test_optimize`function is missing one bracket closing at end accoding to flake, can you exaplain how you declared `args`.

**Comment by araffin** [comment]
>OKAY and what about assigned but never used variables in hyperparams?

if you don't point to line, I cannot answer...

> can you explain why the assert has these lines?

yes, without that assert, if you write incorrect argument in the YAML file, it will fail with a weird error.
This is meant to be more informative.

>login in if statement, we can create a function with that code and then call that function in if statement just to reduce complexity

could you elaborate?

>is missing one bracket closing at end accoding to flake

Yes, I just integrated flake but did not solve the issues yet.

Please open a PR with the files for the different questions (otherwise, it is hard to follow)

**Comment by salmannotkhan** [comment]
Okay, Creating a draft PR.

**Comment by araffin** [comment]
@salmannotkhan I will fix the remaining warnings from the linter.
What will be missing is to use type hints everywhere...
